ouroboros-network/src/Ouroboros/Network/MockChain/Chain.hs

```diff
@@ -232,7 +232,9 @@ selectChain
   -> Chain block
   -> Chain block
 selectChain c1 c2 =
-  if headBlockNo c1 >= headBlockNo c2
+  -- Cannot use headBlockNo, because the block number could be less than the
+  -- length (it is non-striclty increasing).
+  if length c1 >= length c2
```

Comment:
This particular change may not be a good idea and I'm thinking it should be reverted. There's a choice to be made here: if a chain has more blocks, but a smaller block number, is it better? Supposing `c1` and `c2` are Byron chains (including EBBs), where `c1` ends in an EBB for the 2nd epoch, but `c2` ends in a main block in the 1st epoch, and restricted to the 1st epoch, `c2` has 1 more main block than `c1`, we'd want to take `c2`. If we do it on chain length like this, `c1` would still be considered best. But if we do it on block number, `c2` would win, because the EBB in `c1` has the same block number as its parent, which is 1 less than that of `c2`.

Reply:
Yes I think the block number is the right thing to do chain selection on. It doesn't actually matter than we don't prefer `X; EBB` over `X`, because we will only encounter these during syncing where there's always a later block eventually. Once we're involved in consensus and creating blocks, then there will be no EBBs.